File: LAB0/LAB0d.py
def get_heights(n: int) -> list[int]:
    arr: list[int] = [int(x) for x in input().split()]
    return arr[:n]

def get_total(h: list[int], w: int, i: int, j: int) -> int:
    total: int = 0
    total += sum(h[i: j+1])
    total += (j-i+1)*w*2
    return total

t: int = int(input())
for _ in range(t):
    h: list[int] = get_heights(int(input()))
    q: int = int(input())
    waters: int = 0
    for __ in range(q):
        command: list[str] = input().split()
        if command[0] == 'water':
            waters += 1
        elif command[0] == 'total':
            print(get_total(h, waters, int(command[1])-1, int(command[2])-1))

"""
3 1 4 1 5 9
14
total 2 4

"""

File: LAB0/LAB0d.c
#include <stdio.h>
#include <string.h>

void magtanim(long long int n, long long int* h, long long int q){
    long long int prefix[n+1];
    prefix[0] = 0;
    for (long long int i = 1; i <= n; i++) {
        prefix[i] = prefix[i-1] + h[i-1];
    }

    long long int waters = 0;
    for (long long int q_ = 0; q_ < q; q_++) {
        char command[5];
        scanf("%s", command);
        if (strcmp(command, "water") == 0) {
            waters++;
        } else if (strcmp(command, "total") == 0) {
            long long int i, j;
            scanf("%lld %lld", &i, &j);
            long long int total = ((j-i+1)*waters*2) + (prefix[j]-prefix[i]+h[i-1]);
            printf("%lld\n", total);
        }
    }
}


long long int main() {
    long long int t;
    scanf("%lld", &t);

    for (long long int t_ = 0; t_ < t; t_++) {
        long long int n;
        scanf("%lld", &n);
        long long int h[n];
        for (long long int n_ = 0; n_ < n; n_++) {
            long long int temp;
            scanf("%lld", &temp);
            h[n_] = temp;
        }
        long long int q;
        scanf("%lld", &q);
        magtanim(n, h, q);
    }
}